### PR TITLE
Require redis-cli for testing

### PR DESCRIPTION
**Pull request by torwig**
Without `redis-cli` the command `./x.py test go` is running but the test cases eventually fail with an error `exec: "redis-cli": executable file not found in $PATH`

### CODE REVIEW

**Review by PragmaTwice — DISMISSED** [review]

**Review by git-hulk — DISMISSED** [review]

**Review by PragmaTwice — DISMISSED** [review]

**Review by tisonkun — CHANGES_REQUESTED** [review]
We should respect potentially passed arg `cliPath`.

https://github.com/apache/incubator-kvrocks/blob/9665bbfd72f32a32a2bdde90ee48f84998d5e8cb/tests/gocase/util/flags.go#L27

It just happens to be default to `redis-cli`.

File: x.py [hunk 1]
```diff
@@ -266,6 +266,7 @@ def test_cpp(dir: str, rest: List[str]) -> None:
 
 def test_go(dir: str, rest: List[str]) -> None:
     go = find_command('go', msg='go is required for testing')
+    find_command('redis-cli', msg='redis-cli is required for testing')
```
Comment (PragmaTwice) [inline x.py]:
```suggestion
def test_go(dir: str, cli_path: str, rest: List[str]) -> None:
    go = find_command('go', msg='go is required for testing')
    find_command(cli_path, msg='redis-cli is required for testing')
```

To respect `cliPath` according to @tisonkun , you can follow these proposed changes:

And add this line to `__main__`:
```
parser_test_go.add_argument('--cli-path', default='redis-cli', help="path of redis-cli to test kvrocks")
```

Append `cliPath` to `args`:
```
args = [
        'test', '-bench=.', './...',
        f'-binPath={binpath}',
        f'-cliPath={cli_path}',
        f'-workspace={worksapce}',
        *rest
    ]
```

**Comment by torwig** [comment]
@tisonkun Good observation.
@PragmaTwice Thank you for your hints.
Done.

**Comment by torwig** [comment]
Also reformatted `x.py` in a separate commit.

**Review by tisonkun — APPROVED** [review]

**Comment by tisonkun** [comment]
@torwig may I ask what tools you use to format the python code?

**Comment by torwig** [comment]
@tisonkun It was just `Pycharm` (IDE) :)

**Review by PragmaTwice — APPROVED** [review]

**Comment by PragmaTwice** [comment]
Thanks all. Merging...

**Pull request merged**